docs/language/functions.rst

```diff
@@ -279,6 +279,39 @@ constructor.
     for storage rent and there is a minimum deposit needed for the contract. As a result,
     constructors always receive value on Parity Substrate.
 
+Overriding function selector
+____________________________
+
+When a function is called, the function selector and the arguments to the function
+are serialized (also known as abi encoded) and passed to the program. The selector
+is what the runtime program uses to determine what function was called. Usually
+the function selector is generated from a deterministic hash value of the function
+name and its arguments.
+
+To change the selector value, it can be overriden with the ``selector=hex"deadbea1"``
+syntax, for example:
+
+.. code-block:: solidity
+
+    contract foo {
+        function get_foo() selector=hex"01020304" public return (int) {
+            return 102;
+        }
+    }
+
+Only ``public`` and ``external`` functions have a selector, and can have their
+selector overriden. On Substrate, constructors have selectors too, so they have
+can also have their selector overriden. If a function overrides a function in a
+base contract, then the selector of the function in the base contract must match.
+
+.. warning::
+    On ewasm and Solana, changing the selector may result in a mismatch between
+    the contract metadata and the actual contract code. This is because the ewasm
+    and Solana metadata does not explicitly store the selector, but derives it
```

Comment:
```suggestion
    On ewasm and Solana, changing the selector may result in a mismatch between
    the contract metadata and the actual contract code, because the former does not explicitly store the selector, but derives it
```

Reply:
A suggestion to make it less wordy.